apps/dotcom/sync-worker/src/worker.ts

```diff
@@ -195,6 +197,21 @@ export default class Worker extends WorkerEntrypoint<Environment> {
 			throw err
 		})
 	}
+
+	override async queue(batch: MessageBatch<QueueMessage>): Promise<void> {
+		const db = createPostgresConnectionPool(this.env, 'sync-worker-queue')
+		for (const message of batch.messages) {
+			const { objectName, fileId } = message.body
+			try {
+				await db.insertInto('asset').values({ objectName, fileId }).executeTakeFirstOrThrow()
+				message.ack()
+			} catch (_e) {
+				message.retry({
+					delaySeconds: QUEUE_BASE_DELAY ** message.attempts,
```

Comment:
should we put some max limit on this? like after 10 attempts maybe we should stop trying altogether, because the file id is most likely wrong? It's unlikely that the db would be down for 2^10 seconds (~20 minutes) at least.

Reply:
i guess we already validated the file id by this point 🤔 

still technically possible we could hard delete a file and cause this to get in an infinite failure loop.

maybe stop after 12 attempts or something?

Reply:
Yeah, no pointing in retying into perpetuity 😁

Added a `max_retries` option and a dead letter queue for production:
https://github.com/tldraw/tldraw/pull/6027/commits/11d613e4e33a0be074de3851ba53eb0a3fc4311e